js/main.js

```diff
@@ -62,23 +67,730 @@ function getWinLosses(id) {
 
 function idSubmit(event) {
   event.preventDefault();
+
+  if (data.id !== null) {
+    var $oldMatches = document.querySelectorAll('.matches-row');
+    for (var i = 0; i < $oldMatches.length; i++) {
+      $oldMatches[i].remove();
+
+    }
+  }
   data.id = $form.elements.steamid.value;
   getSteamProfile(data.id);
   getWinLosses(data.id);
+  getMatches(data.id);
   $homeView.className = 'home-view hidden';
   $profileView.className = 'profile-view';
   data.view = 'profile';
 
 }
 
+function getMatches(id) {
+
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', 'https://api.opendota.com/api/players/' + id + '/recentMatches');
+  xhr.responseType = 'json';
+  xhr.addEventListener('load', function () {
+
+    var $table = document.querySelector('.table-matches');
+
+    for (var i = 0; i < xhr.response.length; i++) {
+
+      var $tableRow = document.createElement('tr');
+      $tableRow.className = 'matches-row';
+      $table.appendChild($tableRow);
+
+      var $heroImgCol = document.createElement('td');
+      $tableRow.appendChild($heroImgCol);
+
+      var $heroImg = document.createElement('img');
+
+      var $heroCol = document.createElement('td');
+
+      for (var x = 0; x < heroData.length; x++) {
+        if (heroData[x].id === xhr.response[i].hero_id) {
+          $heroImg.setAttribute('src', heroData[x].image);
+          $heroCol.textContent = heroData[x].name_loc;
+          break;
+        }
+      }
+      $heroImgCol.appendChild($heroImg);
+
+      $tableRow.appendChild($heroCol);
+
+      var $resultCol = document.createElement('td');
+      if (xhr.response[i].radiant_win === true) {
+        $resultCol.textContent = 'Radiant Victory';
+      } else {
+        $resultCol.textContent = 'Dire Victory';
+      }
+      $tableRow.appendChild($resultCol);
+
+      var $typeCol = document.createElement('td');
+      if (xhr.response[i].game_mode === 22) {
+        $typeCol.textContent = 'All Pick';
+      } else {
+        $typeCol.textContent = 'Custom Game';
+      }
+      $tableRow.appendChild($typeCol);
+
+      var $durationCol = document.createElement('td');
+      var duration = xhr.response[i].duration / 60;
+      var difference = duration - Math.floor(duration);
+      var seconds = 60 * difference;
+      var secondsFixed = seconds.toFixed(0);
+      if (secondsFixed.toString().length === 1) {
+        $durationCol.textContent = Math.floor(duration) + ':0' + secondsFixed;
+
+      } else {
+        $durationCol.textContent = Math.floor(duration) + ':' + secondsFixed;
+      }
+      $tableRow.appendChild($durationCol);
+
+      var $kdaCol = document.createElement('td');
+      $kdaCol.textContent = xhr.response[i].kills + '/' + xhr.response[i].deaths + '/' + xhr.response[i].assists;
+      $tableRow.appendChild($kdaCol);
```

Comment:
You should make this its own named function with a parameter of an object with the data necessary to generate this table row, then call that function here with `xhr.response[i]` as an argument